docs/apis/streaming/windows.md

```diff
@@ -24,1023 +24,593 @@ specific language governing permissions and limitations
 under the License.
 -->
 
+Flink uses a concept called *windows* to divide a (potentially) infinite `DataStream` into slices
+based on the timestamps of elements or other criteria. This division is required when working
+with infinite streams of data and performing transformations that aggregate elements.
+
+<span class="label label-info">Info</span> We will mostly talk about *keyed windowing* here, this
+means that the elements are subdivided based on both window and key before being given to
+a user function. Keyed windows have the advantage that work can be distributed across the cluster
+because the elements for different keys can be processed in isolation. If you absolutely must,
+you can check out [non-keyed windowing](#non-keyed-windowing) where we describe how non-keyed
+windows work.
+
 * This will be replaced by the TOC
 {:toc}
 
-## Windows on Keyed Data Streams
-
-Flink offers a variety of methods for defining windows on a `KeyedStream`. All of these group elements *per key*,
-i.e., each window will contain elements with the same key value.
+## Basics
 
-### Basic Window Constructs
+For a windowed transformations you must at least specify a *key* (usually in the form of a
```

Comment:
Saying `KeySelector` here is a bit tricky because it implies that users can only specify an actual `KeySelector` while they can also specify an index or a field name, i.e. `keyBy(1,3)` or `keyBy("word")`.